internal/server/cookie.go

```diff
@@ -2,42 +2,66 @@ package server
-	CookieLoginName         = "login"
-	CookieDomain            = ""
-	CookiePath              = "/"
-	// while these vars look goofy, they avoid "magic number" arguments to SetCookie
-	CookieHTTPOnlyNotJavascriptAccessible = true    // setting HttpOnly to true means JS can't access it.
-	CookieSecureHTTPSOnly                 = true    // setting Secure to true means the cookie is only sent over https connections
-	CookieMaxAgeDeleteImmediately         = int(-1) // <0: delete immediately
-	CookieMaxAgeNoExpiry                  = int(0)  // zero has special meaning of "no expiry"
+const (
+	cookieAuthorizationName       = "auth"
+	cookieLoginName               = "login"
+	cookiePath                    = "/"
+	cookieMaxAgeDeleteImmediately = -1 // <0: delete immediately
+	cookieMaxAgeNoExpiry          = 0  // zero has special meaning of "no expiry"
 )
 
 func setAuthCookie(c *gin.Context, key string, expires time.Time) {
 	maxAge := int(time.Until(expires).Seconds())
-	if maxAge == CookieMaxAgeNoExpiry {
-		maxAge = CookieMaxAgeDeleteImmediately
+	if maxAge == cookieMaxAgeNoExpiry {
+		maxAge = cookieMaxAgeDeleteImmediately
 	}
 
-	secure := CookieSecureHTTPSOnly
+	secure := true
 	if c.Request.TLS == nil {
 		// if the request came over HTTP, then the cookie will need to be sent unsecured
 		secure = false
 	}
 
-	c.SetSameSite(http.SameSiteStrictMode)
-
-	c.SetCookie(CookieAuthorizationName, key, maxAge, CookiePath, CookieDomain, secure, CookieHTTPOnlyNotJavascriptAccessible)
-	c.SetCookie(CookieLoginName, "1", maxAge, CookiePath, CookieDomain, secure, CookieHTTPOnlyNotJavascriptAccessible)
+	http.SetCookie(c.Writer, &http.Cookie{
+		Name:     cookieAuthorizationName,
+		Value:    url.QueryEscape(key),
+		MaxAge:   maxAge,
+		Path:     cookiePath,
+		SameSite: http.SameSiteStrictMode,
+		Secure:   secure,
+		HttpOnly: true, // not accessible by javascript
+	})
+	http.SetCookie(c.Writer, &http.Cookie{
+		Name:     cookieLoginName,
+		Value:    "1",
+		MaxAge:   maxAge,
+		Path:     cookiePath,
+		SameSite: http.SameSiteStrictMode,
+		Secure:   secure,
+		HttpOnly: true, // not accessible by javascript
+	})
 }
 
-func deleteAuthCookie(c *gin.Context) {
-	c.SetCookie(CookieAuthorizationName, "", CookieMaxAgeDeleteImmediately, CookiePath, CookieDomain, CookieSecureHTTPSOnly, CookieHTTPOnlyNotJavascriptAccessible)
-	c.SetCookie(CookieLoginName, "", CookieMaxAgeDeleteImmediately, CookiePath, CookieDomain, CookieSecureHTTPSOnly, CookieHTTPOnlyNotJavascriptAccessible)
+func deleteAuthCookie(writer http.ResponseWriter) {
+	http.SetCookie(writer, &http.Cookie{
+		Name:     cookieAuthorizationName,
+		MaxAge:   cookieMaxAgeDeleteImmediately,
+		Path:     cookiePath,
+		Secure:   true, // only over https
+		HttpOnly: true, // not accessible by javascript
+	})
+
+	http.SetCookie(writer, &http.Cookie{
+		Name:     cookieLoginName,
+		MaxAge:   cookieMaxAgeDeleteImmediately,
+		Path:     cookiePath,
+		Secure:   true, // only over https
```

Comment:
I believe this needs to be false when http

Reply:
Let's discuss in the other comment thread: https://github.com/infrahq/infra/pull/2702#discussion_r932621182

I left some links there